### PR TITLE
Seabeta 312 feedback survey

### PR DESCRIPTION
**Pull request by CodrinCotarlanSoftwire**

### CODE REVIEW

File: SeaPublicWebsite/Controllers/FeedbackController.cs [hunk 1]
```diff
@@ -1,10 +1,11 @@
-﻿using Microsoft.AspNetCore.Mvc;
+﻿using System.Linq;
+using Microsoft.AspNetCore.Mvc;
 using SeaPublicWebsite.ExternalServices.EmailSending;
 using SeaPublicWebsite.Models.Feedback;
 
 namespace SeaPublicWebsite.Controllers;
 
-[Route("feedback")]
```
Comment (CodrinCotarlanSoftwire) [inline SeaPublicWebsite/Controllers/FeedbackController.cs:7]: Decided to use the same controller for both feedback forms

File: SeaPublicWebsite/Views/Shared/_Layout.cshtml [hunk 2]
```diff
@@ -81,28 +81,34 @@
 
     }
 
-    <div class="govuk-width-container">
-        
-        @if (ViewBag.ServiceUnavailable != true)
```
Comment (CodrinCotarlanSoftwire) [inline SeaPublicWebsite/Views/Shared/_Layout.cshtml:86]: This wasn't set _anywhere_

File: SeaPublicWebsite/Views/Shared/_Layout.cshtml [hunk 3]
```diff
@@ -81,28 +81,34 @@
...
-    </div>
+<div class="govuk-width-container">
+    
+    @(await Html.GovUkPhaseBanner(new PhaseBannerViewModel
+    {
+        Phase = "Beta",
+        Html = @<text>
+                   This is a new service – your
+                   @{
+                       if (ViewBag.OnFeedbackPage is true)
```
Comment (CodrinCotarlanSoftwire) [inline SeaPublicWebsite/Views/Shared/_Layout.cshtml:92]: ViewBag :(
Works quite well in this case, though

File: SeaPublicWebsite/Controllers/FeedbackController.cs [hunk 4]
```diff
@@ -34,6 +35,39 @@ public IActionResult FeedbackForm_Post(FeedbackFormViewModel viewModel)
             viewModel.WhatUserToldUs);
         return RedirectToAction(nameof(FeedbackThankYou_Get), "Feedback");
     }
+    
+    [HttpGet("feedback-survey")]
+    public IActionResult FeedbackSurvey_Get()
+    {
+        var viewModel = new FeedbackSurveyViewModel();
+        ViewBag.OnFeedbackPage = true;
```
Comment (DanCorderSoftwire) [inline SeaPublicWebsite/Controllers/FeedbackController.cs:43]: Should we have this on the `FeedbackForm_Get` method too?

File: SeaPublicWebsite/Controllers/FeedbackController.cs [hunk 5]
```diff
@@ -34,6 +35,39 @@ public IActionResult FeedbackForm_Post(FeedbackFormViewModel viewModel)
...
+        return View("FeedbackSurvey", viewModel);
+    }
+
+    [HttpPost("feedback-survey")]
+    public IActionResult FeedbackSurvey_Post(FeedbackSurveyViewModel viewModel)
+    {
+        if (!viewModel.VisitReasonList.Any())
+        {
+            ModelState.AddModelError(nameof(viewModel.VisitReasonList), "Please select at least one option");
```
Comment (DanCorderSoftwire) [inline SeaPublicWebsite/Controllers/FeedbackController.cs:52]: I don't think these error messages match the Gov UK style (see https://design-system.service.gov.uk/components/checkboxes/)
e.g. this one should be something like "Select at least one reason that you came to the site today"

File: SeaPublicWebsite/Controllers/FeedbackController.cs [hunk 6]
```diff
@@ -34,6 +35,39 @@ public IActionResult FeedbackForm_Post(FeedbackFormViewModel viewModel)
...
+        {
+            ModelState.AddModelError(nameof(viewModel.HowInformationHelpedList), "Please select at least one option");
+        }
+        if (!viewModel.WhatPlannedToDoList.Any())
+        {
+            ModelState.AddModelError(nameof(viewModel.WhatPlannedToDoList), "Please select at least one option");
+        }
+        if (!ModelState.IsValid)
+        {
+            ViewBag.OnFeedbackPage = true;
```
Comment (DanCorderSoftwire) [inline SeaPublicWebsite/Controllers/FeedbackController.cs:64]: This should be `return FeedbackSurvey_Get()`?
Then you won't need to set the ViewBag either.

Comment (DanCorderSoftwire) [reply SeaPublicWebsite/Controllers/FeedbackController.cs:64]: Can you fix `FeedbackForm_Post` at the same time please?

File: SeaPublicWebsite/Views/Feedback/FeedbackSurvey.cshtml [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+        OverrideWithJavascript = true
+    })
+}
+
+<div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+        <form action="@Url.Action("FeedbackSurvey_Post", "Feedback")" method="post" novalidate>
+            @Html.AntiForgeryToken()
+            
+            <fieldset class="govuk-fieldset">
```
Comment (DanCorderSoftwire) [inline SeaPublicWebsite/Views/Feedback/FeedbackSurvey.cshtml:22]: I'm not sure using a `fieldset` and `legend` here makes sense?
Can you just use the `h1` and `p` on their own?

**Review by DanCorderSoftwire — CHANGES_REQUESTED** [review]

**Review by DanCorderSoftwire — APPROVED** [review]

**Pull request merged**